### DISCUSSION TITLE
Can't run Scalene

### DISCUSSION THREAD

**Discussion by manluckk** [plasma-umass/scalene#449] [Q&A]
I have a problem not being able to run Scalene on my Macbook Air via terminal (bash). Here are the details of the problem I'm having:
![Screen Shot 2022-10-17 at 23 59 06(2)](https://user-images.githubusercontent.com/50958036/196367470-0bfb12c5-cd85-456a-bac6-eb14a0fdde12.png)
Details code of main.py:
![Screen Shot 2022-10-18 at 14 33 17](https://user-images.githubusercontent.com/50958036/196365798-80038713-94ab-466e-a32b-991080102270.png)
My python version:
![Screen Shot 2022-10-18 at 14 28 24](https://user-images.githubusercontent.com/50958036/196365113-d9ec7cce-3309-41ae-b15b-87e770a4dc4d.png)
When I try to run Scalene on Google Colab, I get the same results as when I run it on the terminal (bash). Here is the screenshot:
![Screen Shot 2022-10-17 at 23 07 29](https://user-images.githubusercontent.com/50958036/196366886-3d17c9fa-ca31-4df7-a608-c974e427bd79.png)
My list of pip freeze:
![pipfreeze](https://user-images.githubusercontent.com/50958036/196371690-13889b51-f303-48be-a47c-a7849503f80f.png)
This is my Macbook Air specification:
- Macbook Air 2017
- macOS High Sierra version 10.13.6
- Processor Intel Core i5-5350U @ 1.80Ghz
- Memory 8 GB 1600Mhz DDR3
- Intel HD Graphics 6000 1536 MB
---------------
When I tried on a personal computer with Windows 10 64bit, I was able to run Scalene via Command Prompt (cmd). However, the results I get when running Scalene are not as complete as what is exemplified. Source: [1](https://nbviewer.org/github/plasma-umass/scalene/blob/master/docs/scalene-demo.ipynb); [2](https://github.com/plasma-umass/scalene/tree/v1.5.13#output); and [3](https://github.com/plasma-umass/scalene/tree/v1.5.13#web-based-gui).

A screenshot of the results of running Scalene via Command Prompt (cmd):
![2022-10-18 13_51_45-](https://user-images.githubusercontent.com/50958036/196374095-a770d175-6c24-4ff4-ba44-4f059bd758a2.png)

Here is a screenshot when I'm running Scalene on Google Colab on my PC with Windows 10:
![2022_10_18_13_54_37_](https://user-images.githubusercontent.com/50958036/196374444-332158d3-b2e8-4df2-941e-5baeae93b78f.png)

My personal computer specification:
- Processor: Intel(R) Core(TM) i5-4460  CPU @ 3.20GHz   3.20 GHz
- Installed RAM: 8,00 GB (7,90 GB usable)
- System type: 64-bit operating system, x64-based processor
- Edition: Windows 10 Pro
- Version: 22H2
- OS build: 19045.2075
- Experience: Windows Feature Experience Pack 120.2212.4180.0

**Comment by emeryberger** [comment]
Can you try running Scalene in a directory where there are no spaces? It looks like your Downloads directory has for some reason a bunch of spaces before and after the word "Downloads", and Scalene (and Python in general, AFAICT) does not properly handle this case.

**Reply by manluckk** [reply]
After I'm trying your advice, my problem can't run Scalene on Macbook Air is solved.
But, when I'd run it on my PC with Windows 10, shows messages in the below picture. What must I do?
![2022_11_29_05_10_30_Window](https://user-images.githubusercontent.com/50958036/204466124-f7215bf6-b885-4979-a4bf-3ed0cfbf0ae5.png)

**Reply by emeryberger** [reply]
That message is Mac/Linux specific; for Windows, you should be able to resolve it by running Scalene once in Administrator mode. (I have not tested this on Windows, as I have no Windows w/NVIDIA GPU at my disposal, so please update.)

**Reply by manluckk** [reply]
When I try to run Scalene via Windows Powershell or Windows Command Prompt as Administrator, the result is the same: no tests were performed by Scalene on my Python files,

On my PC, I have an NVIDIA GPU card from Zotac with the model GTX 750 and the newest driver version 526.98.

Can it be concluded that Scalene cannot be used on Windows with GPUs from NVIDIA?


![1111](https://user-images.githubusercontent.com/50958036/204696711-a2245181-74e1-4992-95b4-63aaa576b9c2.png)
